packages/richtext-slate/src/field/elements/EnabledRelationshipsCondition.tsx

```diff
@@ -1,18 +1,28 @@
 'use client'
 
+import type { User } from 'payload/auth'
 import type { SanitizedCollectionConfig } from 'payload/types'
 
-import { useConfig } from 'payload/components/utilities'
+import { useAuth, useConfig } from 'payload/components/utilities'
 import * as React from 'react'
 
-type options = { uploads: boolean }
+type options = {
+  uploads: boolean
+  user: User
+}
 
 type FilteredCollectionsT = (
   collections: SanitizedCollectionConfig[],
   options?: options,
 ) => SanitizedCollectionConfig[]
 const filterRichTextCollections: FilteredCollectionsT = (collections, options) => {
-  return collections.filter(({ admin: { enableRichTextRelationship }, upload }) => {
+  return collections.filter(({ admin: { enableRichTextRelationship, hidden }, upload }) => {
+    if (hidden) {
```

Comment:
I think this should be 

```ts
if (hidden === false) {
}

```

Otherwise, it would always return false if hidden is a function, even if the function in the end would return false.